contrib/src/main/java/com/datatorrent/contrib/kafka/KafkaTopicOffsetManager.java

```diff
@@ -0,0 +1,487 @@
+  {
+    return offsetsMap;
+  }
+
+  /**
+   * @param offsetsMap
+   *          the offsetsMap to set
+   */
+  public void setOffsetsMap(Map<TopicAndPartition, Long> offsetsMap)
+  {
+    this.offsetsMap = offsetsMap;
+  }
+
+  private int retryCount;
+
+  /**
+   * @return the retryCount
+   */
+  public int getRetryCount()
+  {
+    return retryCount;
+  }
+
+  /**
+   * @param retryCount
+   *          the retryCount to set
+   */
+  public void setRetryCount(int retryCount)
+  {
+    this.retryCount = retryCount;
+  }
+
+  private int correlationId;
+
+  /**
+   * @return the correlationId
+   */
+  public int getCorrelationId()
+  {
+    return correlationId;
+  }
+
+  /**
+   * @param correlationId
+   *          the correlationId to set
+   */
+  public void setCorrelationId(int correlationId)
+  {
+    this.correlationId = correlationId;
+  }
+
+  /**
+   * @return the partitionList
+   */
+  public List<TopicAndPartition> getPartitionList()
+  {
+    return partitionList;
+  }
+
+  /**
+   * @param partitionList
+   *          the partitionList to set
+   */
+  public void setPartitionList(List<TopicAndPartition> partitionList)
+  {
+    this.partitionList = partitionList;
+  }
+
+  /**
+   * @return the consumerGroup
+   */
+  public String getConsumerGroup()
+  {
+    return consumerGroup;
+  }
+
+  /**
+   * @param consumerGroup
+   *          the consumerGroup to set
+   */
+  public void setConsumerGroup(String consumerGroup)
+  {
+    this.consumerGroup = consumerGroup;
+  }
+
+  /**
+   * @return the clientId
+   */
+  public String getClientId()
+  {
+    return clientId;
+  }
+
+  /**
+   * @param clientId
+   *          the clientId to set
+   */
+  public void setClientId(String clientId)
+  {
+    this.clientId = clientId;
+  }
+
+  private transient BlockingChannel channel;
+
+  /**
+   * @return the channel
+   */
+  public BlockingChannel getChannel()
+  {
+    return channel;
+  }
+
+  /**
+   * @param channel
+   *          the channel to set
+   */
+  public void setChannel(BlockingChannel channel)
+  {
+    this.channel = channel;
+  }
+
+  public KafkaTopicOffsetManager()
+  {
+
+  }
+
+  public KafkaTopicOffsetManager(String hostname, int port, String consumerGroup, String clientID, String topic,
+      int partitionCount)
+  {
+
+    partitionList = new LinkedList<TopicAndPartition>();
+
+    offsetsMap = new HashMap<TopicAndPartition, Long>();
+
+    setHostname(hostname);
+    setConsumerGroup(consumerGroup);
+    setClientId(clientID);
+    setTopic(topic);
+    setPartitionCount(partitionCount);
+    setPort(port);
+    setRetryCount(retryCount);
+
+    getConnectionHandle(getRetry());
+  }
+
+  public void getConnectionHandle(int retry)
+  {
+
+    if (retry == 0) {
+      setChannel(null);
+      return;
+    }
+
+    int correlationID = 0;
+    try {
+      BlockingChannel channel = new BlockingChannel(getHostname(), getPort(), BlockingChannel.UseDefaultBufferSize(),
```

Comment:
Why not give a list of broker hosts instead of one. Broker can go down, but cluster may still function correctly.